ClientApp/src/components/pages/EmployeeDirectoryPage/EmployeeDirectoryPage.tsx

```diff
@@ -1,43 +1,77 @@
 import {
   AsyncComponent,
+  EditTeamModal,
   ErrorModal,
   Header,
   PermissionsWrapper,
-  useErrorHandling,
 } from "components";
 import { chain, Dictionary } from "lodash";
-import React, { useEffect, useState } from "react";
+import React, { useCallback, useEffect, useState } from "react";
 import Button from "react-bootstrap/Button";
 import Col from "react-bootstrap/Col";
 import Row from "react-bootstrap/Row";
-import { getAllEmployees, useRequireLogin } from "services";
+import { getAllEmployees, useErrorHandling, useRequireLogin } from "services";
+import { useModal } from "utilities";
 import { EmployeeOverview, PermissionType } from "_generated";
 import "./EmployeeDirectoryPage.scss";
 import { EmployeeDirectoryTable } from "./EmployeeDirectoryTable";
 
 export const EmployeeDirectoryPage: React.FC = (): JSX.Element => {
-  const [isLoading, setIsLoading] = useState(true);
+  const [isLoading, setIsLoading] = useState(false);
   const [employees, setEmployees] = useState<EmployeeOverview[]>([]);
   const { error, handleError } = useErrorHandling();
+  const {
+    isModalShowing,
+    setIsModalShowing,
+    setModalData,
+    modalData,
+  } = useModal<EmployeeOverview[]>();
   useRequireLogin();
 
+  /**
+   * Used when sorting and grouping employees.
+   */
+  const getManagerName = (e: EmployeeOverview): string => {
+    if (e.manager?.employeeId === e.employeeId) return "Unassigned";
+    return e.manager?.fullName || "";
+  };
+
+  /**
+   * Note: endpoint was added that sends the employees in teams
+   *  with an unassigned section,
+   *  but then it would require two requests for the same data.
+   * Ideally, this would return a manager: EmployeeOverview as a key
+   *  but I don't know how to do that.
+   */
   const sortAndGroup = (
     emps: EmployeeOverview[],
   ): Dictionary<EmployeeOverview[]> => {
     return chain(emps)
-      .sortBy(
-        (e: EmployeeOverview): string | undefined =>
-          e.manager?.fullName || undefined,
-      )
-      .groupBy(
-        (e: EmployeeOverview): string | undefined =>
-          e.manager?.fullName || undefined,
-      )
+      .sortBy((e: EmployeeOverview): string => getManagerName(e))
+      .groupBy((e: EmployeeOverview): string => getManagerName(e))
       .value();
   };
 
-  useEffect((): (() => void) => {
+  /**
+   * Used to get the manager: EmployeeOverview.
+   * Check that the array is not-undefined,
+   *   the people in the array are not unassigned,
+   *   and then return the manager.
+   *   Else: return an unassigned manager object.
+   */
+  const getManager = (
+    emps: EmployeeOverview[] | undefined,
+  ): EmployeeOverview => {
+    return (
+      (!!emps &&
+        emps[0].employeeId !== emps[0].manager?.employeeId &&
+        emps[0].manager) || { employeeId: -1, fullName: "Unassigned" }
+    );
+  };
```

Comment:
Similar to my previous comment, if there is no manager, it makes more sense to return undefined than a "fake" manager.
```suggestion
  const getManager = (
    emps: EmployeeOverview[] | undefined,
  ): EmployeeOverview | undefined => {
    return (
      (!!emps &&
        emps[0].employeeId !== emps[0].manager?.employeeId &&
        emps[0].manager) || undefined
    );
  };
```

Reply:
Thanks for the comment. With this, I would push back because returning undefined here will just cause more inline ternaries which will make the JSX more complex than needed. I prefer to get rid of potential `null | undefined`  through abstractions. In this part of the logic, all employees who are unassigned are effectively under the same unassigned manager.